vdr/ambassador.go

```diff
@@ -232,13 +232,29 @@ func (n *ambassador) handleUpdateDIDDocument(transaction dag.Transaction, propos
 		WithField(core.LogFieldTransactionRef, transaction.Ref()).
 		WithField(core.LogFieldDID, proposedDIDDocument.ID).
 		Debug("Handling DID document update")
-	// Resolve latest version of DID Document
-	currentDIDDocument, _, err := n.didStore.Resolve(proposedDIDDocument.ID, &types.ResolveMetadata{AllowDeactivated: true})
-	if err != nil {
-		return fmt.Errorf("unable to update DID document: %w", err)
+
+	// Resolve version of DID Document referred to by transaction
+	var currentDIDDocument *did.Document
+	var err error
+	for _, ref := range transaction.Previous() {
+		currentDIDDocument, _, err = n.didStore.Resolve(proposedDIDDocument.ID, &types.ResolveMetadata{AllowDeactivated: true, SourceTransaction: &ref})
```

Comment:
This was my first approach of fixing this issue, but I found out there were 2 issues with it:

- This also returns deactivated documents, which don't contain controllers or keys, which can lead to the same situation (if referring a deactivated document).
- It just takes the first document it finds in the prevs, which might not contain the controller the TX was signed with (e.g. 2 branches of the DID document, the controller being used only present in one of the branches, which isn't referred).

I think the correct way would be to find all prevs and merge the resulting controllers. And I think `resolveControllers` doesn't take all prevs into account as well?

The issue might be too edge case/hard to fix right now. Do we still allow this situation with the multi chain? If not, it might not be worth fixing 100% (although not looking for deactivated DID documents is probably an easy fix).